### DISCUSSION TITLE
No @Query over some methods in a few repositories

### DISCUSSION THREAD

**Discussion by mepox** [trebol-ecommerce/trebol-backend-monolith#166] [Q&A] [answered]
For example in this class: `IUsersJpaRepository` https://github.com/trebol-ecommerce/spring-boot-backend/blob/main/src/main/java/org/trebol/jpa/repositories/IUsersJpaRepository.java

There is no @Query annotation before this method: ```Optional<User> findByName(String name);```

However this method get called for example here in this class: `UsersJpaCrudServiceImpl` https://github.com/trebol-ecommerce/spring-boot-backend/blob/main/src/main/java/org/trebol/jpa/services/crud/UsersJpaCrudServiceImpl.java

In this method: `UsersJpaCrudServiceImpl`:
```
@Override
  public Optional<User> getExisting(UserPojo input) throws BadInputException {
    String name = input.getName();
    if (name == null || name.isBlank()) {
      throw new BadInputException("Invalid user name");
    } else {
      return userRepository.findByName(name);
    }
  }
```

I don't understand how it works without the @Query annotation 😕 

- If there is no @Query then the method does nothing? 
- Then why is it used?
- Or is it unimplemented yet?
- Or just left there by mistake?

**Comment by NyorJa** [comment, marked as answer]
I think its handled by the JPA or Spring Data by general.. 
Check this doc
https://docs.spring.io/spring-data/data-commons/docs/1.6.1.RELEASE/reference/html/repositories.html

Check example 1.4

**Reply by mepox** [reply]
"The JPA module supports defining a query manually as String or have it being derived from the method name."

I see, I didn't know this. 😄 Thanks!

https://docs.spring.io/spring-data/jpa/docs/1.5.0.RELEASE/reference/html/jpa.repositories.html
